doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,889 @@
+    /// Interface for retrieving information about an app catalog without acting on it.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    runtimeclass AppCatalogInfo
+    {
+        /// The app catalog's unique identifier.
+        String Id { get; };
+        /// The name of the app catalog.
+        String Name { get; };
+        /// The type of the app catalog.
+        String Type { get; };
+        /// The argument used when adding the app catalog.
+        String Arg { get; };
+        /// The app catalog's extra data string.
+        String ExtraData { get; };
+        /// The last time that this app catalog was updated.
+        Windows.Foundation.DateTime LastUpdateTime { get; };
+        /// The origin of the app catalog.
+        AppCatalogOrigin Origin { get; };
+        /// The trust level of the app catalog
+        AppCatalogTrustLevel TrustLevel { get; };
+    };
+
+    /// A metadata item of a package version.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    enum PackageVersionMetadata
+    {
+        /// The InstallerType of an installed package
+        InstalledType,
+        /// The Scope of an installed package
+        InstalledScope,
+        /// The system path where the package is installed
+        InstalledLocation,
+        /// The standard uninstall command; which may be interactive
+        StandardUninstallCommand,
+        /// An uninstall command that should be non-interactive
+        SilentUninstallCommand,
+        /// The publisher of the package
+        Publisher,
+        /// The locale of the package
+        Locale,
+    };
+
+    /// IMPLEMENTATION NOTE: IPackageVersion from AppInstallerRepositorySearch.h
+    /// A single package version.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    runtimeclass PackageVersionInfo
+    {
+        /// IMPLEMENTATION NOTE: PackageVersionMetadata fields from AppInstallerRepositorySearch.h
+        /// Gets any metadata associated with this package version.
+        /// Primarily stores data on installed packages.
+        /// Metadata fields may have no value (e.g. packages that aren't installed will not have an InstalledLocation).
+        String GetMetadata(PackageVersionMetadata metadataType);
+        /// IMPLEMENTATION NOTE: PackageVersionProperty fields from AppInstallerRepositorySearch.h
+        String Id { get; };
+        String Name { get; };
+        String Version { get; };
+        String Channel { get; };
+        /// DESIGN NOTE: RelativePath from AppInstallerRepositorySearch.h is excluded as not needed.
+        /// String RelativePath;
+
+        /// IMPLEMENTATION NOTE: PackageVersionMultiProperty fields from AppInstallerRepositorySearch.h
+        /// PackageFamilyName and ProductCode can have multiple values.
+        Windows.Foundation.Collections.IVectorView<String> PackageFamilyName { get; };

```

Comment:
No, doesn't have to be parallel counts. Manifest spec is here: https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv0.1.md I'll link in doc. The PackageFamilyName and ProductCode do have some correlation since they're both under an Installer, so it could still be an array of objects that just has one of the fields blank sometimes, but that's not how it's stored by the already existing internal implementation that this api is exposing so that wouldn't be as easy a change as most of the other api changes.